lib/cli_util.dart

```diff
@@ -37,6 +37,7 @@ Directory getSdkDir([List<String> cliArgs]) {
 
   // Try and locate the VM using 'which'.
   String executable = whichSync('dart', orElse: () => null);
+  if (executable == null) return null;
```

Comment:
What about wrapping the code referencing `executable` below in an `if (executable != null)`? That way we can avoid an early return; might prevent issues w/ future code maintenance.